src/submission-handlers/encounterLocationHandler.test.ts

```diff
@@ -0,0 +1,260 @@
+        encounterContext.encounter,
+        field,
+        [],
+        encounterContext,
+      );
+      expect(initialValue).toEqual('81e6e516-c1f0-11eb-8529-0242ac130003');
+    });
+
+    it('should return undefined if neither the encounter nor the context has a location', () => {
+      // Setup field
+      const field: FormField = {
+        label: 'Encounter Location',
+        type: 'encounterLocation',
+        required: false,
+        id: 'encounterLocation',
+        questionOptions: {
+          rendering: 'ui-select-extended',
+        },
+        validators: [],
+      };
+
+      const initialValue = EncounterLocationSubmissionHandler.getInitialValue(
+        encounterWithoutLocation,
+        field,
+        [],
+        contextWithoutLocation,
+      );
+      expect(initialValue).toBeUndefined();
+    });
+  });
+  describe('getDisplayValue', () => {
+    it('should return display value when value is defined', () => {
+      const field: FormField = {
+        label: 'Encounter Location',
+        type: 'encounterLocation',
+        required: false,
+        id: 'encounterLocation',
+        questionOptions: {
+          rendering: 'ui-select-extended',
+        },
+        validators: [],
+      };
+
+      const value = { display: 'Test Location', uuid: '5c95f6f5-788e-4e73-9079-5626911231fa' };
+      const displayValue = EncounterLocationSubmissionHandler.getDisplayValue(field, value);
+
+      expect(displayValue).toEqual('Test Location');
+    });
+
+    it('should return undefined when value is null', () => {
+      const field: FormField = {
+        label: 'Encounter Location',
+        type: 'encounterLocation',
+        required: false,
+        id: 'encounterLocation',
+        questionOptions: {
+          rendering: 'ui-select-extended',
+        },
+        validators: [],
+      };
+
+      const value = null;
+      const displayValue = EncounterLocationSubmissionHandler.getDisplayValue(field, value);
+
+      expect(displayValue).toBeUndefined();
+    });
+
+    it('should return undefined when value is undefined', () => {
+      const field: FormField = {
+        label: 'Encounter Location',
+        type: 'encounterLocation',
+        required: false,
+        id: 'encounterLocation',
+        questionOptions: {
+          rendering: 'ui-select-extended',
+        },
+        validators: [],
+      };
+
+      const value = undefined;
+      const displayValue = EncounterLocationSubmissionHandler.getDisplayValue(field, value);
+
+      expect(displayValue).toBeUndefined();
+    });
+  });
+  describe('getPreviousValue', () => {
+    it('should return display and value when encounter has location', () => {
+      const field: FormField = {
+        label: 'Encounter Location',
+        type: 'encounterLocation',
+        required: false,
+        id: 'encounterLocation',
+        questionOptions: {
+          rendering: 'ui-select-extended',
+        },
+        validators: [],
+      };
```

Comment:
@samuelmale is this a pattern that will be maintained? The field definition seems to be repeated throughout the file but it's the same content. Or is it this way so each instance in the test is independent?

Reply:
I think for this scenario it should be possible to reuse a single field object across the suite. We can then add a cleanup or teardown callback that [ensures that the side effects from the previous tests are cleaned up](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/components/inputs/select/dropdown.test.tsx#L78-L81)

Reply:
thanks @pirupius @samuelmale ..
i have added a tear down